xwiki-rendering-syntaxes/xwiki-rendering-syntax-xhtml/src/main/java/org/xwiki/rendering/internal/parser/xhtml/wikimodel/XWikiMacroHandler.java

```diff
@@ -62,6 +70,24 @@ public XWikiMacroHandler(ComponentManager componentManager, XHTMLParser parser)
         this.parser = parser;
     }
 
+    private String getSyntax(TagContext context) throws ComponentLookupException
```

Comment:
I think I would deal with the exception where getInstance is called instead and fallback on parser syntax.